tests/src/test/java/alluxio/server/configuration/ConfigCheckerIntegrationTest.java

```diff
@@ -0,0 +1,150 @@
+        .build();
+    try {
+      cluster.start();
+      MetaMasterClient client = cluster
+          .waitForAllNodesRegistered(30 * Constants.SECOND_MS);
+      Assert.assertEquals(TEST_NUM_MASTERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_MASTER_NUM)))
+          .getConfMasterNum());
+      cluster.notifySuccess();
+    } finally {
+      cluster.destroy();
+    }
+  }
+
+  @Test
+  public void MultiWorkers() throws Exception {
+    Map<Integer, Map<PropertyKey, String>> workerProperties
+        = generateProperties(TEST_NUM_WORKERS, PropertyKey.Scope.WORKER);
+    MultiProcessCluster cluster = MultiProcessCluster.newBuilder()
+        .setClusterName("ConfigCheckerMultiWorkersTest")
+        .setNumMasters(1)
+        .setNumWorkers(TEST_NUM_WORKERS)
+        .setWorkerProperties(workerProperties)
+        .build();
+
+    try {
+      cluster.start();
+      MetaMasterClient client = cluster
+          .waitForAllNodesRegistered(30 * Constants.SECOND_MS);
+
+      Assert.assertEquals(1, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_MASTER_NUM)))
+          .getConfMasterNum());
+      Assert.assertEquals(TEST_NUM_WORKERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_WORKER_NUM)))
+          .getConfWorkerNum());
+      cluster.notifySuccess();
+    } finally {
+      cluster.destroy();
+    }
+  }
+
+  @Test
+  public void MultiNodes() throws Exception {
+    // Prepare properties
+    Map<Integer, Map<PropertyKey, String>> properties = generateProperties(
+        TEST_NUM_MASTERS + TEST_NUM_WORKERS, PropertyKey.Scope.SERVER);
+    Map<Integer, Map<PropertyKey, String>> masterProperties = properties.entrySet().stream()
+        .filter(entry -> (entry.getKey() < TEST_NUM_MASTERS))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    Map<Integer, Map<PropertyKey, String>> workerProperties = properties.entrySet().stream()
+        .filter(entry -> (entry.getKey() >= TEST_NUM_MASTERS))
+        .collect(Collectors.toMap(entry -> entry.getKey() - TEST_NUM_MASTERS, Map.Entry::getValue));
+
+    MultiProcessCluster cluster = MultiProcessCluster.newBuilder()
+        .setClusterName("ConfigCheckerMultiNodesTest")
+        .setNumMasters(TEST_NUM_MASTERS)
+        .setNumWorkers(TEST_NUM_WORKERS)
+        .setDeployMode(MultiProcessCluster.DeployMode.ZOOKEEPER_HA)
+        .setMasterProperties(masterProperties)
+        .setWorkerProperties(workerProperties)
+        .build();
+
+    try {
+      cluster.start();
+      MetaMasterClient client = cluster.waitForAllNodesRegistered(
+          30 * Constants.SECOND_MS);
+
+      Assert.assertEquals(TEST_NUM_MASTERS, client.getMasterInfo(
+          new HashSet<>(Arrays.asList(MasterInfo.MasterInfoField.CONF_MASTER_NUM)))
+          .getConfMasterNum());
+      Assert.assertEquals(TEST_NUM_WORKERS, client.getMasterInfo(
```

Comment:
in all of these tests, can we check that the config checker report contains exactly the right data?

Reply:
The fsadmin doctor has been merged, so I add the check report codes. Mainly check the config status and if the key that has different values exists in the corresponding warns or errors. Thanks